src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php

```diff
@@ -2657,21 +2657,42 @@ private function parseAlternativeScopeTermination($tokenType)
     private function parseExpressionList(ASTNode $exprList)
     {
         $this->consumeComments();
-        while ($expr = $this->parseOptionalExpression()) {
-            $exprList->addChild($expr);
 
-            $this->consumeComments();
-            if (Tokens::T_COMMA === $this->tokenizer->peek()) {
-                $this->consumeToken(Tokens::T_COMMA);
-                $this->consumeComments();
-            } else {
-                break;
-            }
-        }
+        while (
+            ($expr = $this->parseOptionalExpression()) &&
+            $this->addChildToList($exprList, $expr)
+        );
```

Comment:
I find do-while more readable since it breaks up the condition and doesn't look like it's doing nothing :)
```suggestion
        do (
            $expr = $this->parseOptionalExpression();
        ) while($expr && $this->addChildToList($exprList, $expr));
```